pkg/compact/dedup/dedup.go

```diff
@@ -0,0 +1,79 @@
+package dedup
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+type BucketDeduper struct {
```

Comment:
We are missing comments on exported structs / methods / functions.

Reply:
I will review all the codes and see whether I can change some exported structs / methods / functions to interval. If not, I will add comments for them. Thanks.